### DISCUSSION TITLE
opencv-python version installed every time the webui is started

### DISCUSSION THREAD

**Discussion by tmprabubiz** [Mikubill/sd-webui-controlnet#2107] [Q&A]
Python 3.10.6 (tags/v3.10.6:9c7b4bd, Aug  1 2022, 21:53:49) [MSC v.1932 64 bit (AMD64)]
Version: v1.6.0
Commit hash: 5ef669de080814067961f28357256e8fe27544f4
Installing sd-webui-controlnet requirement: changing opencv-python version from 4.7.0.72 to 4.8.0
Checking roop requirements
Install insightface==0.7.3
Installing sd-webui-roop requirement: insightface==0.7.3
Install onnx==1.14.0
Installing sd-webui-roop requirement: onnx==1.14.0
Install onnxruntime==1.15.0
Installing sd-webui-roop requirement: onnxruntime==1.15.0
Install opencv-python==4.7.0.72
Installing sd-webui-roop requirement: opencv-python==4.7.0.72
Launching Web UI with arguments: --xformers --medvram-sdxl --no-half-vae --no-gradio-queue --theme dark --autolaunch
[-] ADetailer initialized. version: 23.9.2, num models: 9
2023-09-12 21:11:41,712 - ControlNet - INFO - ControlNet v1.1.410
ControlNet preprocessor location: C:\Users\Winnertmp\a1111\stable-diffusion-webui\extensions\sd-webui-controlnet\annotator\downloads
2023-09-12 21:11:41,896 - ControlNet - INFO - ControlNet v1.1.410
2023-09-12 21:11:42,607 - roop - INFO - roop v0.0.2
2023-09-12 21:11:42,730 - roop - INFO - roop v0.0.2
Loading weights [700528894b] from C:\Users\Winnertmp\a1111\stable-diffusion-webui\models\Stable-diffusion\juggernautXL_version2.safetensors
Creating model from config: C:\Users\Winnertmp\a1111\stable-diffusion-webui\repositories\generative-models\configs\inference\sd_xl_base.yaml
Running on local URL:  http://127.0.0.1:7860
_______________________________________________________________________________________________
Controlnet requires opencv-python version 4.8.0
sd-webui-roop requires opencv-python version 4.7.0.72
So during start both versions are loaded one by one and the start up time too much.
What should I do Please guide.

**Comment by satangel2222** [comment]
same issue here need help

Python 3.10.6 (main, Dec 22 2022, 15:39:53) [MSC v.1934 64 bit (AMD64)]
Version: v1.5.2
Commit hash: c9c8485bc1e8720aba70f029d25cba1c4abf2b5c

Installing sd-webui-controlnet requirement: changing opencv-python version from 4.7.0.72 to 4.8.0

Checking roop requirements
Install insightface==0.7.3
Installing sd-webui-roop requirement: insightface==0.7.3
Install onnx==1.14.0
Installing sd-webui-roop requirement: onnx==1.14.0
Install onnxruntime==1.15.0
Installing sd-webui-roop requirement: onnxruntime==1.15.0
Install opencv-python==4.7.0.72
Installing sd-webui-roop requirement: opencv-python==4.7.0.72

loading WD14-tagger reqs from E:\stable-diffusion-webui\extensions\stable-diffusion-webui-wd14-tagger\requirements.txt
Checking WD14-tagger requirements.

**Reply by grahamquest** [reply]
What I did was just change the requirements.txt for roop to be 4.8...tadah! 
(please tell me if this is a dumb idea)

**Reply by tmprabubiz** [reply]
A big "NO"
I tried your advice, but much more errors arise in command box.
There should be a perfect method to rectify the issue.

**Reply by grahamquest** [reply]
damn! It worked for me a little bit, but no youre right... it destroyed it ahahah. OK a big NO it is!

**Comment by huchenlei** [comment]
If dw_pose works in `4.7.0.72`, we can change the requirement version to `>=4.7.0.72`. The opencv version dependency was added because dw_pose inference fails in older version of opencv.

**Comment by ultimatech-cn** [comment]
I changed the requirements.txt of roop as following. 
```
insightface>=0.7.3
onnx>=1.14.0
onnxruntime>=1.15.0
opencv-python>=4.7.0.72
ifnude
cython
```

**Reply by tmprabubiz** [reply]
Already my  requirements.txt of roop exists the same

**Comment by Luvdraft** [comment]
Did anyone find a solution to this?

**Comment by bhadrik** [comment]
I found out this command line argument `--skip-install` this is given in [command-line-wiki](https://github.com/AUTOMATIC1111/stable-diffusion-webui/wiki/Command-Line-Arguments-and-Settings) It works perfectly fine for me.

### **In case you don't know here is how to use:**
1. Go to your `stable-diffusion-webui` folder
2. Open `webui-user.bat` file with notepad or any text editor. Your file should contain text similar to shown below. No need to be exact same.
     

```
@echo off

set PYTHON=
set GIT=
set VENV_DIR=
set COMMANDLINE_ARGS= --skip-install --xformers --api --listen --no-half-vae

call webui.bat
```
3. Now add `--skip-install` as shown in above file, make sure you add this into `set COMANDLINE_ARGS line`.
4. That's all it should work ;)

**Reply by Arijit0** [reply]
Not working bro

**Comment by Nakasara1** [comment]
i have the same problem here too, but i want to use controlnet. what i noticed is Automatic1111 works at all, but controlnet has no impact on the picture.

So how do i update this to 4.8.0 manually?

**Reply by pkj2008456** [reply]
so did u fix it? I can't fix it :(

**Comment by vaulthunt3r** [comment]
You need to find requirements.txt in which 4.8.0 is written and replace it with 4.7.0.72. I encountered the same problem on new versions
